future/transaction.go

```diff
@@ -528,3 +529,355 @@ func byte32FromBase64(in string) (out [32]byte, err error) {
 	copy(out[:], slice)
 	return
 }
+
+// - accounts     lists the accounts (in addition to the sender) that may be accessed
+//                from the application logic.
+//
+// - appArgs      ApplicationArgs lists some transaction-specific arguments accessible
+//                from application logic.
+//
+// - appIdx       ApplicationID is the application being interacted with, or 0 if
+//                creating a new application.
+//
+// - approvalProg ApprovalProgram determines whether or not this ApplicationCall
+//                transaction will be approved or not.
+//
+// - clearProg    ClearStateProgram executes when a clear state ApplicationCall
+//                transaction is executed. This program may not reject the
+//                transaction, only update state.
+//
+// - foreignApps  lists the applications (in addition to txn.ApplicationID) whose global
+//                states may be accessed by this application. The access is read-only.
+// - globalSchema GlobalStateSchema sets limits on the number of strings and
+//                integers that may be stored in the GlobalState. The larger these
+//                limits are, the larger minimum balance must be maintained inside
+//                the creator's account (in order to 'pay' for the state that can
+//                be used). The GlobalStateSchema is immutable.
+//
+// - localSchema  LocalStateSchema sets limits on the number of strings and integers
+//                that may be stored in an account's LocalState for this application.
+//                The larger these limits are, the larger minimum balance must be
+//                maintained inside the account of any users who opt into this
+//                application. The LocalStateSchema is immutable.
+//
+// - onComplete   This is the faux application type used to distinguish different
+//                application actions. Specifically, OnCompletion specifies what
+//                side effects this transaction will have if it successfully makes
+//                it into a block.
+
+// MakeApplicationCreateTx makes a transaction for creating an application (see above for args desc.)
+func MakeApplicationCreateTx(
+	onComplete types.OnCompletion,
```

Comment:
We shouldn't accept `OnCompletion` here, the only valid `OnCompletion` types are `NoOpOC` and `OptInOC`.

I think that the options are either (1) remove the field, or (2) change the field to a boolean `optInSender` (or a better name).